src/main/java/org/apache/flink/benchmark/UnalignedCheckpointTimeBenchmark.java

```diff
@@ -42,119 +43,90 @@
+                        .build();
 
         new Runner(options).run();
     }
 
     @Benchmark
-    public void unalignedCheckpoint(UnalignedCheckpointEnvironmentContext context) throws Exception {
+    public void unalignedCheckpoint(UnalignedCheckpointEnvironmentContext context)
+            throws Exception {
         StreamExecutionEnvironment env = context.env;
-        DataStreamSource<byte[]> source = env.addSource(new FiniteCheckpointSource(NUM_FINISHED_CHECKPOINTS));
-        source
-            .slotSharingGroup("source").rebalance()
-            .map((MapFunction<byte[], byte[]>) value -> value).slotSharingGroup("map").rebalance()
-            .addSink(new SlowDiscardSink<>()).slotSharingGroup("sink");
+        DataStreamSource<Record> source =
+                env.fromSource(
+                        new RecordSource(NUM_FINISHED_CHECKPOINTS),
+                        noWatermarks(),
+                        RecordSource.class.getName());
+
+        source.slotSharingGroup("source")
+                .rebalance()
+                .map((MapFunction<Record, Record>) value -> value)
+                .slotSharingGroup("map")
+                .rebalance()
+                .addSink(new SlowDiscardSink<>())
+                .slotSharingGroup("sink");
 
         env.execute();
     }
 
     public static class UnalignedCheckpointEnvironmentContext extends FlinkEnvironmentContext {
 
-        @Param({"REMOTE", "LOCAL"})
-        public String mode = "REMOTE";
+        @Param({"0", "1", "ALIGNED"})
+        public String timeout = "0";
 
         @Setup
         public void setUp() throws IOException {
             super.setUp();
 
             env.setParallelism(parallelism);
             env.enableCheckpointing(CHECKPOINT_INTERVAL_MS);
-            env.getCheckpointConfig().enableUnalignedCheckpoints(true);
-            env.getCheckpointConfig().setAlignmentTimeout(Duration.ZERO);
+            if ("ALIGNED".equals(timeout)) {
+                env.getCheckpointConfig().enableUnalignedCheckpoints(false);
+            } else {
+                env.getCheckpointConfig().enableUnalignedCheckpoints(true);
+                env.getCheckpointConfig()
+                        .setAlignmentTimeout(Duration.ofMillis(Integer.parseInt(timeout)));
+            }
         }
 
         protected Configuration createConfiguration() {
             Configuration conf = super.createConfiguration();
-
-            if (mode.equals("REMOTE")) {
-                conf.setInteger(TaskManagerOptions.NUM_TASK_SLOTS, 1);
-                conf.setInteger(ConfigConstants.LOCAL_NUMBER_TASK_MANAGER, NUM_VERTICES * PARALLELISM);
-            } else {
-                conf.setInteger(TaskManagerOptions.NUM_TASK_SLOTS, NUM_VERTICES * PARALLELISM);
-                conf.setInteger(ConfigConstants.LOCAL_NUMBER_TASK_MANAGER, 1);
-            }
+            conf.setInteger(TaskManagerOptions.NUM_TASK_SLOTS, 1);
```

Comment:
Don't we want to test different numbers of slots per task manager? Or it doesn't make a lot of sense in this scenario?

Reply:
With infinite resources available it would make sense to test with various number of local/remote input channels. But having as many as possible remote input channels and single local channel I think is a worst case scenario that's testing both code paths at the same time, so single slot per TM I think should be good enough, and should cover all cases well enough.

If needed, we can expand this test in the future.